test-d/foundry/common/documents.mjs/baseActor.test-d.ts

```diff
@@ -0,0 +1,82 @@
+import { expectError, expectType } from 'tsd';
+import '../../../../index';
+import EmbeddedCollection from '../../../../src/foundry/common/abstract/embedded-collection.mjs';
+import { PropertiesToSource } from '../../../../src/types/helperTypes';
+import { ActiveEffectDataProperties } from '../../../../src/foundry/common/data/data.mjs/activeEffectData';
+import { EffectDurationDataProperties } from '../../../../src/foundry/common/data/data.mjs/effectDurationData';
+
+const baseActor = new foundry.documents.BaseActor();
+expectType<EmbeddedCollection<typeof ActiveEffect, foundry.data.ActorData>>(baseActor.effects);
+expectType<EmbeddedCollection<typeof Item, foundry.data.ActorData>>(baseActor.items);
+expectType<PropertiesToSource<ActiveEffectDataProperties>>(baseActor.data._source.effects[0]);
+expectType<EffectDurationDataProperties | undefined>(baseActor.data._source.effects[0].duration);
+
+interface CharacterDataSourceData {
+  health: number;
+}
+
+interface CharacterDataSource {
+  type: 'character';
+  data: CharacterDataSourceData;
+}
+
+interface CharacterDataPropertiesData extends CharacterDataSource {
```

Comment:
```suggestion
interface CharacterDataPropertiesData extends CharacterDataSourceData {
```
This makes the test green ;)